consensus/state.go

```diff
@@ -674,9 +674,28 @@ func (cs *State) updateToState(state sm.State) {
 		// to be gathered for the first block.
 		// And alternative solution that relies on clocks:
 		// cs.StartTime = state.LastBlockTime.Add(timeoutCommit)
-		cs.StartTime = cs.config.Commit(cmttime.Now())
+
+		if state.LastBlockHeight == 0 {
```

Comment:
Isn't `CommitTime.IsZero()` only for the first block i.e. when `state.LastBlockHeight == 0`. Can you find a case where this is not true?

Reply:
If you want to keep this can you compact this into a simpler `IsGenesis` method and then just have a single if statement to set `cs.StartTime`